### PR TITLE
#4 Add Dockerfile and docker-compose

### PR DESCRIPTION
**Pull request by SergKam**
- Dockerfile and docker-compose added as requested in #4 
-  `nullPassportStrategy.js` added to bypass authentication,
- Some workaround was required to deal with Nakadi issue https://github.com/zalando/nakadi/issues/792 
- package.json changed and  package-lock.json added to reduce warnings on docker image compiling.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+FROM node:latest as builder
```
Comment (lmontrieux) [inline Dockerfile]: I thought that using latest wasn't allowed by the rules of play?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+WORKDIR /src
+
+COPY . .
+RUN npm install
+RUN npm run build
+
+
+FROM node:latest
+MAINTAINER Sergii Kamenskyi <sergukam@sergukam.com>
+LABEL Description="This Nakadi UI do not check auth by default. Please use Auth plugins for use in production."
```
Comment (lmontrieux) [inline Dockerfile]: `do not` -> `does not`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      POSTGRES_PASSWORD: nakadi
+      POSTGRES_DB: local_nakadi_db
+
+  zookeeper:
+    image: wurstmeister/zookeeper:3.4.6
+    ports:
+      - "2181:2181"
+
+  kafka:
+    image: wurstmeister/kafka:0.10.1.0
```
Comment (lmontrieux) [inline docker-compose.yml]: We use a more recent version of kafka in nakadi now :)

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+COPY webpack.config.prod.js webpack.config.prod.js
+COPY elm-package.json elm-package.json
+RUN npm install && npm run build
+
+FROM registry.opensource.zalan.do/stups/node:8.9.4-alpine-34
+MAINTAINER Sergii Kamenskyi <sergukam@sergukam.com>
+LABEL Description="This Nakadi UI does not check auth by default. Please use Auth plugins for use in production."
+
+WORKDIR /app
+COPY --from=builder /src/dist dist
```
Comment (adyach) [inline Dockerfile]: It would be nice to copy only one time, if possible, for example `COPY --from=builder /src/ .`

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+COPY --from=builder /src/server server
+COPY --from=builder /src/package.json .
+RUN npm install --only=production
+
+EXPOSE 3000/tcp
+
+ENV HTTP_PORT="3000"
```
Comment (adyach) [inline Dockerfile]: I prefer to put env var to env file or docker compose file. That way Dockefile looks cleaner.

File: Dockerfile (hunk 4 above)
Comment (SergKam) [reply Dockerfile]: the src dir contains a lot of trash (build system modules, tmp build artifacts, etc ) that we don't need.
I want to have the smallest and cleanest container possible.

File: Dockerfile (hunk 5 above)
Comment (SergKam) [reply Dockerfile]: Yes, it is possible. But I want to show to the reader of the Dockerfile what options are possible, it is easier to see here without documentation.

**Pull request merged**